src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +34,260 @@ public void GetSerializableMembers_Interface()
+            Assert.Throws<SerializationException>(() => FormatterServices.GetSafeUninitializedObject(nonRuntimeType));
+        }
+#endif
+
+        public static IEnumerable<object[]> GetUninitializedObject_NotSupportedType_TestData()
+        {
+            yield return new object[] { typeof(int[]) };
+            yield return new object[] { typeof(int[,]) };
+            yield return new object[] { typeof(int).MakePointerType() };
+            yield return new object[] { typeof(int).MakeByRefType() };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_NotSupportedType_TestData))]
+        public void GetUninitializedObject_NotSupportedType_ThrowsArgumentException(Type type)
+        {
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [InlineData(typeof(AbstractClass))]
+        [InlineData(typeof(StaticClass))]
+        [InlineData(typeof(Interface))]
+        [InlineData(typeof(Array))]
+        public void GetUninitializedObject_AbstractClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        private abstract class AbstractClass { }
+        private static class StaticClass { }
+        private interface Interface { }
+
+        public static IEnumerable<object[]> GetUninitializedObject_OpenGenericClass_TestData()
+        {
+            yield return new object[] { typeof(GenericClass<>) };
+            yield return new object[] { typeof(GenericClass<>).MakeGenericType(typeof(GenericClass<>)) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+
+            yield return new object[] { typeof(Span<int>) };
```

Comment:
The full framework runtime is not aware of Span type special properties today, so this test will fail on full framework.